File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientPartPure.java
package com.mytlogos.enterprise.background.api.model;

import java.util.Arrays;

/**
 * API Model for MinPart.
 */
public class ClientPartPure {
    private final int mediumId;
    private final int id;
    private final String title;
    private final int totalIndex;
    private final int partialIndex;

    public ClientPartPure(int mediumId, int id, String title, int totalIndex, int partialIndex) {
        this.mediumId = mediumId;
        this.id = id;
        this.title = title;
        this.totalIndex = totalIndex;
        this.partialIndex = partialIndex;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ClientPartPure that = (ClientPartPure) o;

        return getId() == that.getId();
    }

    @Override
    public int hashCode() {
        return getId();
    }

    @Override
    public String toString() {
        return "ClientPart{" +
                "mediumId=" + mediumId +
                ", id=" + id +
                ", title='" + title + '\'' +
                ", totalIndex=" + totalIndex +
                ", partialIndex=" + partialIndex +
                '}';
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public int getTotalIndex() {
        return totalIndex;
    }

    public int getPartialIndex() {
        return partialIndex;
    }

    public int getMediumId() {
        return mediumId;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientPartPure.kt
package com.mytlogos.enterprise.background.api.model

import com.mytlogos.enterprise.model.Indexable

/**
 * API Model for MinPart.
 */
class ClientPartPure(
    val mediumId: Int,
    val id: Int,
    val title: String,
    override val totalIndex: Int,
    override val partialIndex: Int,
) : Indexable {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ClientPartPure
        return id == that.id
    }

    override fun hashCode(): Int {
        return id
    }

    override fun toString(): String {
        return "ClientPart{" +
                "mediumId=" + mediumId +
                ", id=" + id +
                ", title='" + title + '\'' +
                ", totalIndex=" + totalIndex +
                ", partialIndex=" + partialIndex +
                '}'
    }
}